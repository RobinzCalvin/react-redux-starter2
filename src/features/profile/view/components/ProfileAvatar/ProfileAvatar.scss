.profile-avatar {
  border-radius: 50%;
  background-position: 50%;

  &_size {
    &_big {
      width: 10rem;
      height: 10rem;
    }

    &_small {
      width: 2.5rem;
      height: 2.5rem;
      border: 0.1rem solid #e5e5e5;
    }
  }
}
