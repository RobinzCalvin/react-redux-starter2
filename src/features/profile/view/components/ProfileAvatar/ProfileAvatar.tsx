import React from 'react';
import block from 'bem-cn';

import './ProfileAvatar.scss';

interface IProps {
  avatarURL: string;
  size: 'small' | 'big';
}

const b = block('profile-avatar');

function ProfileAvatar(props: IProps) {
  const { avatarURL, size } = props;
  return (
    <div className={b({ size })} style={{ backgroundImage: `url(${avatarURL})` }} />
  );
}

export { ProfileAvatar };
