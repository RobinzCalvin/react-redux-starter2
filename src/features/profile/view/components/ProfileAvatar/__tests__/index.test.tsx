import React from 'react';
import { shallow } from 'enzyme';

import ProfileAvatar from '../ProfileAvatar';

describe('ProfileAvatar component', () => {
  it('should match snapshot', () => {
    const component = shallow(<ProfileAvatar size="big" avatarURL="" />);
    expect(component).toMatchSnapshot();
  });
});
