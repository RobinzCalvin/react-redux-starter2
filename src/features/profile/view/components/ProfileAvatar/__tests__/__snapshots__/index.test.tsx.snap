// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileAvatar component should match snapshot 1`] = `
<div
  className={
    Object {
      "has": [Function],
      "is": [Function],
      "mix": [Function],
      "split": [Function],
      "state": [Function],
      "toString": [Function],
    }
  }
  style={
    Object {
      "backgroundImage": "url()",
    }
  }
/>
`;
